translation/message_selector.go

```diff
@@ -0,0 +1,239 @@
+package translation
+
+import (
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+type MessageSelector struct{}
+
+func NewMessageSelector() *MessageSelector {
+	return &MessageSelector{}
+}
+
+// Choose a translation string from an array according to a number.
+func (m *MessageSelector) Choose(message string, number int, locale string) string {
+	segments := strings.Split(strings.Trim(message, "\" "), "|")
+	if value := m.extract(segments, number); value != nil {
+		return strings.Trim(*value, " ")
+	}
+
+	segments = stripConditions(segments)
+	pluralIndex := getPluralIndex(number, locale)
+
+	if len(segments) == 1 || pluralIndex >= len(segments) {
+		return strings.Trim(segments[0], "\" ")
+	}
+
+	return segments[pluralIndex]
+}
+
+func (m *MessageSelector) extract(segments []string, number int) *string {
+	for _, segment := range segments {
+		if line := m.extractFromString(segment, number); line != nil {
+			return line
+		}
+	}
+
+	return nil
+}
+
+func (m *MessageSelector) extractFromString(segment string, number int) *string {
```

Comment:
I want to support cases like `{0}|{1}second`. here when number is 0 it should be returning "" zero string. So i need something else which can represent zero value that's why i used pointer so now  i can use value != nil (it will also support "").